data-collection/src/main/java/refactoringml/util/CKUtils.java

```diff
@@ -1,18 +1,45 @@
 package refactoringml.util;
 
+import com.github.mauricioaniche.ck.CK;
 import com.github.mauricioaniche.ck.CKClassResult;
 import com.github.mauricioaniche.ck.CKMethodResult;
+import com.github.mauricioaniche.ck.CKNotifier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import refactoringml.ProcessMetricsCollector;
 import refactoringml.db.ClassMetric;
 import refactoringml.db.MethodMetric;
-
-import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
+import java.util.concurrent.*;
 import java.util.stream.Collectors;
 
+import static refactoringml.util.PropertiesUtils.getProperty;
 import static refactoringml.util.RefactoringUtils.cleanMethodName;
 
 public class CKUtils {
+	private static final Logger log = LogManager.getLogger(ProcessMetricsCollector.class);
+	private static long timeout = Long.parseLong(getProperty("timeoutRefactoringMiner"));
+
+	//Calls the CK.calculate with a timeout.
+	public static void calculate(String tempdir, String commitHash, String projectUrl, CKNotifier ckNotifier){
+		FutureTask timeoutTask = null;
```

Comment:
I think this would only work if JDT constantly checks for `Thread.isInterrupt()`. The JVM itself doesn't kill the thread. The only way to know, I guess, is by testing it.

Reply:
I changed it to an [ExecutionService](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html) which can be shut down.